crates/server/src/rpc/warp_ws_transport.rs

```diff
@@ -23,62 +23,84 @@ impl WarpWebSocketTransport {
-                    }
-                    return Ok(message);
+                    let message_data = message.into_bytes();
+                    return Ok(message_data);
                 } else {
+                    if message.is_close() {
+                        return Err(TransportError::Closed);
+                    }
                     // Ignore messages that are not binary
-                    return Err(TransportError::Internal);
+                    error!("> WebSocketTransport > Received message is not binary");
+                    return Err(TransportError::NotBinaryMessage);
                 }
             }
             Some(Err(err)) => {
-                println!("Failed to receive message {err:?}");
+                error!(
+                    "> WebSocketTransport > Failed to receive message {}",
+                    err.to_string()
+                );
+                if let Some(transport_error) = translate_warp_error(&err) {
+                    Err(transport_error)
+                } else {
+                    Err(TransportError::Internal(Box::new(err)))
+                }
             }
             None => {
-                println!("No message")
+                error!("> WebSocketTransport > None received > Closing...");
+                return Err(TransportError::Closed);
             }
         }
-        println!("Closing transport...");
-        self.close().await;
-        Ok(TransportEvent::Close)
     }
 
     async fn send(&self, message: Vec<u8>) -> Result<(), TransportError> {
         let message = WarpWSMessage::binary(message);
-        self.write
-            .lock()
-            .await
-            .send(message)
-            .await
-            .map_err(|_| TransportError::Internal)?;
-        Ok(())
+        match self.write.lock().await.send(message).await {
+            Err(err) => {
+                error!(
+                    "> WebSocketTransport > Error on sending in a ws connection {}",
+                    err.to_string()
+                );
+                if let Some(transport_error) = translate_warp_error(&err) {
+                    Err(transport_error)
+                } else {
+                    Err(TransportError::Internal(Box::new(err)))
+                }
+            }
+            Ok(_) => Ok(()),
+        }
     }
 
     async fn close(&self) {
         match self.write.lock().await.close().await {
             Ok(_) => {
-                self.ready.store(false, Ordering::SeqCst);
+                debug!("> WebSocketTransport > Closed successfully")
             }
-            _ => {
-                println!("Couldn't close tranport")
+            Err(err) => {
+                error!("> WebSocketTransport > Error: Couldn't close tranport: {err:?}")
             }
         }
     }
+}
 
-    fn is_connected(&self) -> bool {
-        self.ready.load(Ordering::Relaxed)
+fn translate_warp_error(err: &warp::Error) -> Option<TransportError> {
```

Comment:
WDYT if we implement the `From` trait here and instead of returning the `None` variant, we wrap it in a `TransportError::Internal(..)`?

Reply:
We cannot implement the `From` trait here because `TransportError` is a type from the `dcl-rpc` crate